.github/workflows/dependency-updates-post-processing.yaml

```diff
@@ -71,31 +73,48 @@ jobs:
           # An alternative would be to use a tag for this. But this does affect the whole
           # PR instead of just the latest commit.
           declare -r tidy_message='🤖🦾🛠️ go mod tidy & make package-specs'
+          declare -r commit_message='${{steps.configure_git.outputs.commit_subject}}'
 
-          declare -r commit_author_name="${{steps.configure_git.outputs.commit_author_name}}"
-          declare -r commit_message="${{steps.configure_git.outputs.commit_subject}}"
-
-          if [[ "${commit_message}" == "${tidy_message}" ]]; then
-            echo 'This commit is already an automatic `go-mod-tidy and make package-specs`! Exiting ...'
+          if [[ "${commit_message}" == "${tidy_message}" ]]
+          then
+            # shellcheck disable=SC2016
+            echo 'This commit is already an automatic `go-mod-tidy and make package-specs`! Exiting …'
             exit 0
           fi
 
           # Generated files are needed for `go mod tidy` which is a dependency of the
           # target `package-specs`. However the generation of them itself already
           # requires go-modules to be tidied up. So we need to generate the files
           # before changing `go.mod` and `go.sum`.
-          declare -r current_branch="${{ github.head_ref }}"
+          current_branch_head="${PR_BRANCH_REF}"
+          readonly current_branch_head
+          echo "Working in branch \`${current_branch_head}\`" \
+               " on revision \`$(git rev-parse 'HEAD')\`."
+
+          echo "Generating clients, servers and fakes on \`${current_branch_head}~1\`"
+          # Sometimes just loading an env may trigger a change regarding the `plugin_version` of
+          # the derivation python.
+          git restore devbox.lock
           git checkout 'HEAD~1'
+
           make generate-fakes
           make generate-openapi-generated-clients-and-servers
-          git checkout "${current_branch}"
+
+          echo "Successfully generated the code. Switching back to \`${current_branch_head}\`!"
+          # Sometimes just loading an env may trigger a change regarding the `plugin_version` of
+          # the derivation python.
+          git restore devbox.lock
```

Comment:
Would it be possible to skip this restore for now?

We currently rely on this job to fix up renovate devbox PR's, see https://github.com/renovatebot/renovate/discussions/35149

Reply:
I was not in the past, but let's try again!

Reply:
Sorry, no, see <https://github.com/cloudfoundry/app-autoscaler-release/actions/runs/14354647335/job/40241195165?pr=3687>!